include/rmm/mr/device/failure_callback_resource_adaptor.hpp

```diff
@@ -55,7 +55,7 @@ using failure_callback_t = std::function<bool(std::size_t, void*)>;
  * When implementing a callback function for allocation retry, care must be taken to avoid an
  * infinite loop. The following example makes sure to only retry the allocation once:
  *
- * @code{c++}
+ * @code{.cpp}
```

Comment:
Does it work to just leave off the `{.cpp}`? By default, Doxygen uses the language of the file the `@code` is found in. Will that work for the unified case?

Reply:
Unfortunately doxygen-generated XML does not correctly identify `hpp` files as C++ files. It does work for `cpp` files, but since rmm's files are all headers I needed to specify this. I do wonder if updating to a newer version of doxygen would improve the situation, and I may test that. I think #1317 needs merging before an y PR to update doxygen, so if the test is promising I'll sequence that work appropriately.

Reply:
Try modifying the Doxyfile EXTENSION_MAPPING setting?

```
EXTENSION_MAPPING      = cu=C++ \
                         cuh=C++ \
                         hpp=C++
```

Reply:
That doesn't seem to work, but I'm not sure why. Oddly, specifying `@code{.hpp}` works (whether or not hpp is added to the `EXTENSION_MAPPING`), but the default inference of just `@code` doesn't.